depthai_nodes/ml/messages/img_detections.py

```diff
@@ -40,44 +60,150 @@ def keypoints(
+
+        @return: Height of the bounding box.
+        @rtype: float
+        """
+        return self._height
+
+    @property
+    def angle(self) -> float:
+        """Returns the angle of the bounding box.
+
+        @return: Angle of the bounding box.
+        @rtype: float
+        """
+        return self._angle
+
+    @x_center.setter
+    def x_center(self, value: float):
+        """Sets the X coordinate of the center of the bounding box.
+
+        @param value: X coordinate of the center of the bounding box.
+        @type value: float
+        @raise TypeError: If the X coordinate is not a float.
+        """
+        self._x_center = value
+
+    @y_center.setter
+    def y_center(self, value: float):
+        """Sets the Y coordinate of the center of the bounding box.
+
+        @param value: Y coordinate of the center of the bounding box.
+        @type value: float
+        @raise TypeError: If the Y coordinate is not a float.
+        """
+        self._y_center = value
+
+    @width.setter
+    def width(self, value: float):
+        """Sets the width of the bounding box.
+
+        @param value: Width of the bounding box.
+        @type value: float
+        @raise TypeError: If the width is not a float.
+        """
+        self._width = value
+
+    @height.setter
+    def height(self, value: float):
+        """Sets the height of the bounding box.
+
+        @param value: Height of the bounding box.
+        @type value: float
+        @raise TypeError: If the height is not a float.
+        """
+        self._height = value
+
+    @angle.setter
+    def angle(self, value: float):
+        """Sets the angle of the bounding box.
+
+        @param value: Angle of the bounding box.
+        @type value: float
+        @raise TypeError: If the angle is not between -360 and 360.
+        """
+        if value < -360 or value > 360:
+            raise TypeError("Angle must be between -360 and 360 degrees.")
+        self._angle = value
+
+    @property
+    def confidence(self) -> float:
+        """Returns the confidence of the detection.
+
+        @return: Confidence of the detection.
+        @rtype: float
+        """
+        return self._confidence
+
+    @confidence.setter
+    def confidence(self, value: float):
+        """Sets the confidence of the detection.
+
+        @param value: Confidence of the detection.
+        @type value: float
+        """
+        self._confidence = value
+
+    @property
+    def label(self) -> int:
+        """Returns the label of the detection.
+
+        @return: Label of the detection.
+        @rtype: int
+        """
+        return self._label
 
+    @label.setter
+    def label(self, value: int):
+        """Sets the label of the detection.
 
-class ImgDetectionsExtended(dai.ImgDetections):
+        @param value: Label of the detection.
+        @type value: int
+        """
+        self._label = value
+
+
+class ImgDetectionsExtended(dai.Buffer):
```

Comment:
Does removing `dai.ImgDetections` as a parent class break something? IIRC this was added so some features could work.

Reply:
When discussing whith dai team, they wanted it to not inherit from dai.ImgDetections. Thats why it's a seperate class.